src/Agent.Worker/Release/ReleaseJobExtension.cs

```diff
@@ -359,5 +359,35 @@ private void LogDownloadFailureTelemetry(IExecutionContext executionContext, Exc
 
             executionContext.AddIssue(issue);
         }
+
+        private IList<AgentArtifactDefinition> FilterArtifactDefintions(IList<AgentArtifactDefinition> agentArtifactDefinitions)
+        {
+            var definitions = new List<AgentArtifactDefinition>();
+            foreach (var agentArtifactDefinition in agentArtifactDefinitions)
+            {
+                if (agentArtifactDefinition.ArtifactType != AgentArtifactType.Custom)
+                {
+                    definitions.Add(agentArtifactDefinition);
+                }
+                else
+                {
+                    string artifactType = string.Empty;
+                    var artifactDetails = JsonConvert.DeserializeObject<Dictionary<string, string>>(agentArtifactDefinition.Details);
+                    if (artifactDetails.TryGetValue("ArtifactType", out artifactType))
+                    {
+                        if (artifactType == null || artifactType.Equals("Build", StringComparison.OrdinalIgnoreCase))
```

Comment:
ArtifactType keyValue is only available in artifacts as extension. If artifact type is Build then we add it to the download list. Else we will skip it